src/Symfony/Component/PasswordHasher/Tests/Hasher/UserPasswordHasherTest.php

```diff
@@ -72,25 +71,43 @@ public function testHash()
         $mockPasswordHasherFactory = $this->createMock(PasswordHasherFactoryInterface::class);
         $mockPasswordHasherFactory->expects($this->any())
             ->method('getPasswordHasher')
-            ->with($this->equalTo($userMock))
+            ->with($user)
             ->willReturn($mockHasher);
 
         $passwordHasher = new UserPasswordHasher($mockPasswordHasherFactory);
 
-        $encoded = $passwordHasher->hashPassword($userMock, 'plainPassword');
+        $encoded = $passwordHasher->hashPassword($user, 'plainPassword');
         $this->assertEquals('hash', $encoded);
     }
 
-    public function testVerify()
+    public function testHashWithPasswordAuthenticatedUser()
```

Comment:
I will remove `expectError()` with the fix. This is just to show the bug and that replacing other user mocks with concrete instance will not affect their behavior to verify the fix will not affect them.